experiment/src/org/labkey/experiment/publish/SampleTypePublishConfirmAction.java

```diff
@@ -0,0 +1,312 @@
+    }
+
+    @Override
+    protected QuerySettings getQuerySettings(SampleTypePublishConfirmForm form)
+    {
+        String queryName = form.getSampleType().getName();
+        QuerySettings qs = new QuerySettings(form.getViewContext(), QueryView.DATAREGIONNAME_DEFAULT, queryName);
+        qs.setSchemaName(SamplesSchema.SCHEMA_NAME);
+        qs.setQueryName(queryName);
+
+        return qs;
+    }
+
+    @Override
+    protected boolean isMismatchedSpecimenInfo(SampleTypePublishConfirmForm form)
+    {
+        return false;
+    }
+
+    @Override
+    protected ActionURL getPublishHandlerURL(SampleTypePublishConfirmForm form)
+    {
+        return urlProvider(ExperimentUrls.class).getLinkToStudyConfirmURL(getContainer(), _sampleType).deleteParameters();
+    }
+
+    @Override
+    protected Dataset.PublishSource getPublishSource(SampleTypePublishConfirmForm form)
+    {
+        return Dataset.PublishSource.SampleType;
+    }
+
+    @Override
+    protected FieldKey getObjectIdFieldKey(SampleTypePublishConfirmForm form)
+    {
+        return FieldKey.fromParts(ROW_ID);
+    }
+
+    @Override
+    protected Map<PublishResultsQueryView.ExtraColFieldKeys, FieldKey> getAdditionalColumns(SampleTypePublishConfirmForm form)
+    {
+        String valueSource = form.getDefaultValueSource();
+        DefaultValueSource defaultValueSource = DefaultValueSource.valueOf(valueSource);
+
+        Map<PublishResultsQueryView.ExtraColFieldKeys, FieldKey> additionalCols = new HashMap<>();
+        additionalCols.put(PublishResultsQueryView.ExtraColFieldKeys.ParticipantId, defaultValueSource.getParticipantIDFieldKey(_participantId));
+        additionalCols.put(PublishResultsQueryView.ExtraColFieldKeys.VisitId, defaultValueSource.getVisitIDFieldKey(_visitId));
+        additionalCols.put(PublishResultsQueryView.ExtraColFieldKeys.Date, defaultValueSource.getVisitIDFieldKey(_visitId));
+        additionalCols.put(PublishResultsQueryView.ExtraColFieldKeys.SourceId, FieldKey.fromParts(
+                ExpMaterialProtocolInputTable.Column.SampleSet.name(),
+                ExpMaterialProtocolInputTable.Column.RowId.name()));
+
+        return additionalCols;
+    }
+
+    @Override
+    protected ActionURL copyToStudy(SampleTypePublishConfirmForm form, Container targetStudy, Map<Integer, PublishKey> dataKeys, List<String> errors)
+    {
+        List<Map<String, Object>> dataMaps = new ArrayList<>();
+        Map<Container, Set<Integer>> rowIdsByTargetContainer = new HashMap<>();
+
+        for (PublishKey publishKey : dataKeys.values())
+        {
+            Map<String, Object> dataMap = new HashMap<>();
+
+            Container targetStudyContainer = targetStudy;
+            if (publishKey.getTargetStudy() != null)
+                targetStudyContainer = publishKey.getTargetStudy();
+            assert targetStudyContainer != null;
+
+            String sourceLSID = new Lsid("Data", _sampleType.getLSID()).toString();
```

Comment:
I think we literally want to use the sample type lsid, the code you have will mutate the lsid namespace and we wont be able to identify the original source.

```suggestion
            String sourceLSID = sampleType.getLSID();
```

Reply:
Use sample type LSID